spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java

```diff
@@ -127,6 +128,22 @@ else if (result != null) {
 		return null;
 	}
 
+	boolean ignoreAddress(InetAddress address) {
+
+		if (this.properties.isUseOnlySiteLocalInterfaces() && !address.isSiteLocalAddress()) {
+			log.trace("Ignoring address: " + address.getHostAddress());
+			return true;
+		}
+	
+		for (String regex : this.properties.getPreferredNetworks()) {
```

Comment:
Look like a bug. 

```java
if (address.getHostAddress().matches(regex) || address.getHostAddress().startsWith(regex))
```

must be replaced by:

```java
if (!address.getHostAddress().matches(regex) && !address.getHostAddress().startsWith(regex))
```

Now I use only `useOnlySiteLocalInterfaces` on production. `prefferedNetworks` is planned to be used in next months.